### PR TITLE
[Feature][Branch] Koa branch implementation.

### PR DESCRIPTION
**Pull request by lucianlature**
This branch uses [Koa2](https://github.com/koajs/koa/tree/v2.x) as web server, instead of Express.
It makes use of `async/await` for the middlewares.
For live development, the Webpack2 dev server is now a `Koa2` instance too.

Security is provided through `helmet`, however there is not (!yet) `hpp` support for Koa, so that piece is missing.

Happy Coding!

### CODE REVIEW

**Comment by ctrlplusb** [comment]
@lucianlature this looks great! Thanks for this effort. :)

One request from my side, and this is my fault for not finishing the "contributor guidelines" yet.  I have given this some thought and I don't want us to customise the README.md between branches.  I would prefer that a separate file is created, for example BRANCH.md that gives a very brief description of the differences contained within the branch compared to master.  It doesn't need to be an essay or anything.

Otherwise everything else looks great.

I would like to add you as a collaborator and make you "the head of the koa branch" if that is okay?

Basically we should merge your branch into a new branch named:
`incubator/http/koa`

We can leave the code in this branch for a period of time until we all feel happy with it's stability level, and then eventually move it into `features/http/koa`.

Sounds like a plan? :)

File: package.json [hunk 1]
```diff
@@ -103,6 +109,8 @@
     "happypack": "3.0.1",
     "html-webpack-plugin": "2.24.1",
     "json-loader": "0.5.4",
+    "koa-flow-declarations": "^0.1.0",
```
Comment (ctrlplusb) [inline package.json]: Please use "exact" versions. i.e. `0.1.0` vs `~0.1.0`.  You'll probably need to kill and rebuild the yarn.lock file.  When using yarn to add dependencies (as is recommended) pass a `--exact` flag to make sure it puts exact version numbers in the package.json dependency lists.

**Comment by lucianlature** [comment]
Sounds like a good plan 👍 

Meanwhile, I have also setups for `glamor` and `styled-components`. Should I?...

**Comment by ctrlplusb** [comment]
Awesomeness. Yes please. :)

**Comment by ctrlplusb** [comment]
Also, please feel free to add your details to the contributors list in package.json (on master)
